src/client.js

```diff
@@ -0,0 +1,205 @@
+const dgram = require('node:dgram')
+const { EventEmitter } = require('node:events')
+
+const { Connection } = require('./connection')
+const { SignalType, SignalStructure } = require('./signalling')
+
+const { getRandomUint64, createPacketData, prepareSecurePacket, processSecurePacket } = require('./util')
+const { PeerConnection } = require('node-datachannel')
```

Comment:
Need to figure out how to not requre native deps at some point. It'd be great if we could get something working with WASM if needed